src/gordon_gcp/plugins/enricher.py

```diff
@@ -49,11 +57,110 @@ class GCEEnricher:
     """
     phase = 'enrich'
 
-    def __init__(self, config, success_channel, error_channel):
+    def __init__(self, config, http_client, success_channel, error_channel):
         self.config = config
         self.success_channel = success_channel
         self.error_channel = error_channel
+        self._http_client = http_client
+        self._logger = logging.getLogger('')
+
+    async def _poll_for_instance_data(self, resource_name, msg_logger):
+        exception = None
+        backoff = 2
+        base_url = f'https://compute.googleapis.com/compute/v1/{resource_name}'
+
+        # Poll until instance data contains external IP information.
+        for attempt in range(1, self.config['retries'] + 1):
+            try:
+                msg_logger.debug(f'Attempt {attempt}: fetching {base_url}')
+                instance_data = await self._http_client.get_json(base_url)
+
+                a_confs = instance_data['networkInterfaces'][0]['accessConfigs']
```

Comment:
This piece of code simply checks for the existence of a nested key. I think assigning this variable is the lesser evil compared to two other alternatives I thought of: adding an `if` block, which adds yet one more level of indentation, or pulling this out into its own tiny function. Any thoughts/advice?

Reply:
Another approach could be an `assert` statement:

```python
assert instance_data['networkInterfaces'][0]['accessConfigs'][0]['natIP'], 'some error message'
```

(maybe shorten that dict lookup somehow)

I don't necessarily have strong opinions either way.

Reply:
Ah, thanks. Makes the intent more explicit :+1: